packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1144,6 +1144,10 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
             this.shouldLogMaxEnqueuedRequestsExceeded = false;
         }
 
+        if (this.maxRequestsPerCrawl === undefined && options.reason === 'enqueueLimit') {
```

Comment:
I understand the intention behind the `maxRequestsPerCrawl` check - you want to make sure we don't log the message if the "global" limit has also been reached. However, calling `this.calculateEnqueuedRequestLimit` and checking if it's `undefined` or non-zero would be more reliable.